src/services/activities/activity_service.ts

```diff
@@ -28,23 +30,39 @@ export class ActivityService {
     let user = await this.userDAO.findById(userId);
     const nextActivityDate = !isNullOrUndefined(user.lastActivitiesSyncedAt)
       ? new Date(new Date(user.lastActivitiesSyncedAt).getTime())
-      : new Date(new Date().getTime() - 1209600 * 1000); // TODO: make this configurable. Currently set to past two weeks.
+      : new Date(new Date().getTime() - DEFAULT_SYNC_RANGE_IN_SECONDS); // TODO: make this configurable. Currently set to past two weeks.
 
     // get all activities from strava after the time activities were last synced.
     const providerActivities = await this.providerDAO.getActivities(
       nextActivityDate,
     );
 
-    let newLastSyncedAt = new Date();
+    let newLastSyncedAt: Date;
+    if (providerActivities.length > 0) {
+      // set to the last activity found
+      newLastSyncedAt =
+        providerActivities[providerActivities.length - 1].startDate;
+    } else {
+      // set to the last time we synced so that next time we sync we can still catch any activities found since
+      newLastSyncedAt = nextActivityDate;
+    }
 
     let activities: Activity[] = [];
     if (providerActivities.length > 0) {
       activities = await this.activityDAO.createMultiple(providerActivities);
     }
 
     // set the lastActivitiesSyncedAt to now.
-    // TODO: We might want to set this to the date of the last activity, just in case a ride hasn't been uploaded to
-    // strava yet when we perform the sync.
+    // TODO: How do we want to handle the case where a user hasn't logged in for an extended period of time?
```

Comment:
@jmoseley make sure this all makes sense.

Reply:
It's odd they don't have a created at field on the activities. That would solve all the issues... We could just always look back a long period of time (like a year) and do and upsert. That has the benefit of consuming any modifications as well, since details about an activity can change.

Alternatively, at least for now, an activity is only valuable during the duration of a challenge. Once the challenge ends we need to declare a winner, and missing data makes that potentially error prone, but we can't force the user the sync.